lib/puppet/type/package.rb

```diff
@@ -384,6 +385,10 @@ def change_to_s(currentvalue, newvalue)
       desc "A read-only parameter set by the package."
     end
 
+    newparam(:package_arch) do
```

Comment:
Since this is a parameter, puppet won't compare the current arch against the desired arch. So if you had `noarch` installed, but wanted `x86_64` puppet would still think it's in-sync. Would that ever happen in practice? If so, we'd need to make this a property, provided we can always retrieve the current `arch` and compare against the desired value.